plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/database/DatabaseNavigatorTree.java

```diff
@@ -840,8 +854,50 @@ public boolean isElementVisible(Viewer viewer, Object element){
             if (!needToMatch) {
                 return true;
             }
-
-            return super.isLeafMatch(viewer, element);
+            String labelText = ((ILabelProvider) ((ContentViewer) viewer).getLabelProvider()).getText(element);
```

Comment:
I'd move the code below to a separate function.
Otherwise `isElementVisible` is super-long and hard to understand..